library/src/main/java/com/bumptech/glide/request/target/NotificationTarget.java

```diff
@@ -29,35 +30,58 @@
+   * @param viewId          The id of the ImageView view that will load the image.
+   * @param remoteViews     RemoteViews object which contains the ImageView that will load the
+   *                        bitmap.
+   * @param notification    The Notification object that we want to update.
+   * @param notificationId  The notificationId of the Notification that we want to load the Bitmap.
    */
   public NotificationTarget(Context context,
       int viewId, RemoteViews remoteViews, Notification notification, int notificationId) {
-    this(context, SIZE_ORIGINAL, SIZE_ORIGINAL, viewId, remoteViews, notification, notificationId);
+    this(context, viewId, remoteViews, notification, notificationId, null);
   }
 
   /**
-   * Constructor using a Notification object and a notificationId to get a handle on the
-   * Notification in order to update it.
+   * Constructor using a Notification object, a notificationId, and a notificationTag to get a
+   * handle on the Notification in order to update it that uses {@link #SIZE_ORIGINAL} as the
+   * target width and height.
    *
-   * @param context        Context to use in the AppWidgetManager initialization.
-   * @param width          Desired width of the bitmap that will be loaded.(Need to be manually put
-   *                       because of RemoteViews limitations.)
-   * @param height         Desired height of the bitmap that will be loaded. (Need to be manually
-   *                       put because of RemoteViews limitations.)
-   * @param viewId         The id of the ImageView view that will load the image.
-   * @param remoteViews    RemoteViews object which contains the ImageView that will load the
-   *                       bitmap.
-   * @param notification   The Notification object that we want to update.
-   * @param notificationId The notificationId of the Notification that we want to load the Bitmap.
+   * @param context         Context to use in the AppWidgetManager initialization.
+   * @param viewId          The id of the ImageView view that will load the image.
+   * @param remoteViews     RemoteViews object which contains the ImageView that will load the
+   *                        bitmap.
+   * @param notification    The Notification object that we want to update.
+   * @param notificationId  The notificationId of the Notification that we want to load the Bitmap.
+   * @param notificationTag The notificationTag of the Notification that we want to load the Bitmap.
+   *                        May be {@code null}.
    */
-  public NotificationTarget(Context context, int width, int height,
-      int viewId, RemoteViews remoteViews, Notification notification, int notificationId) {
+  public NotificationTarget(Context context, int viewId, RemoteViews remoteViews,
+      Notification notification, int notificationId, String notificationTag) {
+    this(context, SIZE_ORIGINAL, SIZE_ORIGINAL, viewId, remoteViews, notification, notificationId,
+      notificationTag);
```

Comment:
I'm a little torn here. The `notify` and `cancel` methods in the Android API break tradition of putting the optional args last, so the order becomes inconsistent with these constructors and even the doc says `(tag, id)`.

I guess we can leave it as is to help understanding the telescoping constructors better. This and the widget target is probably in need of a builder pattern (out of scope for you now).